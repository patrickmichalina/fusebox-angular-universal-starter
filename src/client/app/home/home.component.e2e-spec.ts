import { baseUrl, browser } from '../../../../tools/test/jest.e2e-setup'

describe('Home Page', () => {
  it('should have title', async () => {
    expect.assertions(1)

    const page = browser.goto(`${baseUrl}`)

    const text = await page.evaluate(() => document.title)

    expect(text).toContain('Home - Fusebox Angular Universal Starter')
  })
})
