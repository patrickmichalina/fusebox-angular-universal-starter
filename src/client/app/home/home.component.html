<div class="jumbotron">
  <h1 class="display-3">Hello, Angular Developer!</h1>
  <p class="lead">Welcome to a blazing fast Angular Universal development starter</p>
  <hr class="my-4">
  <h2>Features</h2>
  <ul>
    <li><a href="http://fuse-box.org/">FuseBox</a> bundling (faster than <a href="https://webpack.github.io/" target="_blank">Webpack</a> and <a href="http://jspm.io/" target="_blank">JSPM</a>)</li>
    <li>Fully typed build tools and application using <a href="https://www.typescriptlang.org/" target="_blank">TypeScript</a></li>
    <li>Production and development builds</li>
    <li>Angular testing with <a href="https://facebook.github.io/jest" target="_blank">Jest</a> (faster than Karma)</li>
    <li>Manage your type definitions using @types (easier than using <a href="https://github.com/typings/typings">typings</a>)</li>
    <li>Simple <a href="https://heroku.com" target="_blank">Heroku</a> deployment</li>
    <li>Cookies Support</li>
    <li><a href="https://github.com/google/brotli" target="_blank">Brotli compression</a> with <a href="http://www.gzip.org" target="_blank">gzip</a> fallback</li>
    <li>HttpStateTransfer for caching server responses on client boostrap (no flickering on load)</li>
    <li>Automatic sitemap generation</li>
    <li><a href="https://circleci.com" target="_blank">CircleCI</a> unit testing support</li>
    <li><a href="https://mobile.angular.io" target="_blank">SCSS</a> support for professional grade CSS management</li>
    <li>Full favicon icon generation for multiple devices derived from a single seed image</li>
    <li>SEO support for Title and Meta tags</li>
    <li><a href="http://ogp.me" target="_blank">OG (Open Graph)</a> tags for social sharing</li>
    <li><strong>Coming Soon</strong>: Auth Service + Auth Interface</li>
    <li><strong>Coming Soon</strong>: Hot Module Reloading for faster browser reloads during development</li>
    <li><strong>Coming Soon</strong>: Support for <a href="https://mobile.angular.io" target="_blank">Angular Mobile Toolkit</a> (Service Worker)</li>
    <li><strong>Coming Soon</strong>: <a href="https://angular.io/guide/aot-compiler" target="_blank">Ahead-of-Time (AOT) compilation</a> support</li>
    <li><strong>Coming Soon</strong>: <a href="https://angular-2-training-book.rangle.io/handout/modules/lazy-loading-module.html" target="_blank">Tree-Shaking</a> builds</li>
    <li><strong>Coming Soon</strong>: <a href="https://angular.io/guide/aot-compiler#tree-shaking-1" target="_blank">Lazy Loaded</a> modules</li>
  </ul>
  <p class="lead">
    <a class="btn btn-primary btn-lg" href="https://github.com/patrickmichalina/fusebox-angular-universal-starter" role="button">View Github Repo</a>
  </p>
</div>