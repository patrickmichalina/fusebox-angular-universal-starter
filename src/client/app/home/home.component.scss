:host {
  #hero {
    display: flex;
    flex-direction: column;
    align-items: center;
    h1 {
      margin-bottom: 0;
    }
  }
}
