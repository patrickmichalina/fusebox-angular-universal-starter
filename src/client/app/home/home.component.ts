import { ChangeDetectionStrategy, Component, HostBinding } from '@angular/core'

@Component({
  selector: 'pm-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class HomeComponent {
  @HostBinding('class.card-float-container') containerClass = true
}
