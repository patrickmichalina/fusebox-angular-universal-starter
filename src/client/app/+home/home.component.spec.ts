import { By } from '@angular/platform-browser'
import { RouterTestingModule } from '@angular/router/testing'
import { Angulartics2, Angulartics2GoogleAnalytics, Angulartics2Module } from 'angulartics2'
import { HomeComponent } from './home.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'
import { HomeModule } from './home.module'
import { MdButtonModule, MdCardModule } from '@angular/material'

describe(HomeComponent.name, () => {
  let fixture: ComponentFixture<HomeComponent>

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [HomeModule, RouterTestingModule,
        Angulartics2Module.forRoot([Angulartics2GoogleAnalytics]),
        MdCardModule, MdButtonModule],
      declarations: [TestComponent]
    }).compileComponents()
  }))

  beforeEach(async(() => {
    fixture = TestBed.createComponent(HomeComponent)
  }))

  afterEach(async(() => {
    TestBed.resetTestingModule()
  }))

  it('should match snapshot', async(() => {
    expect(fixture).toMatchSnapshot()
  }))

  it('should compile', async(() => {
    expect(fixture.nativeElement).toBeDefined()
  }))

  it('should track event when link clicked', async(() => {
    const analytics = TestBed.get(Angulartics2) as Angulartics2
    expect(analytics).toBeDefined()
    const links = fixture.debugElement.queryAll(By.css('.lead a'))
    expect(links).toHaveLength(1)
    const link = links[0].nativeElement as HTMLAnchorElement
    expect(link).toBeDefined()
    analytics.eventTrack.subscribe(eventTrack => {
      expect(eventTrack).toBeDefined()
      expect(eventTrack.action).toEqual('ViewRepo')
      expect(eventTrack.properties.eventType).toEqual('click')
    })
    link.click()
  }))
})

@Component({
  selector: 'test-component',
  template: '<pm-home></pm-home>'
})
class TestComponent { }
