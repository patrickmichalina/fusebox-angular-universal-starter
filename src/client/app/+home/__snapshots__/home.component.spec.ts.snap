// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should compile 1`] = `
<pm-home
  containerClass={[Function Boolean]}
>
  <mat-card
    class="mat-card"
  >
    
    
    <h1
      class="display-3"
    >
      Hello, Angular Developer!
    </h1>
    
    
    <p
      class="lead"
    >
      Welcome to a blazing fast Angular Universal development starter
    </p>
    
    
    <a
      angulartics2on="click"
      angularticsevent="ViewRepo"
      class="mat-raised-button"
      color="primary"
      href="https://github.com/patrickmichalina/fusebox-angular-universal-starter"
      mat-raised-button=""
      role="button"
    >
      <span
        class="mat-button-wrapper"
      >
        
      
        <mat-icon
          aria-hidden="true"
          class="fa fa-github mat-icon"
          role="img"
        />
         Github
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
    
    
    <mat-tab-group
      class="mat-tab-group mat-primary"
    >
      <mat-tab-header
        class="mat-tab-header"
      >
        <div
          aria-hidden="true"
          class="mat-tab-header-pagination mat-tab-header-pagination-before mat-elevation-z4 mat-ripple"
          mat-ripple=""
        >
          <div
            class="mat-tab-header-pagination-chevron"
          />
        </div>
        <div
          class="mat-tab-label-container"
        >
          <div
            class="mat-tab-list"
            role="tablist"
          >
            <div
              class="mat-tab-labels"
            >
              
            </div>
            <mat-ink-bar
              class="mat-ink-bar"
            />
          </div>
        </div>
        <div
          aria-hidden="true"
          class="mat-tab-header-pagination mat-tab-header-pagination-after mat-elevation-z4 mat-ripple"
          mat-ripple=""
        >
          <div
            class="mat-tab-header-pagination-chevron"
          />
        </div>
      </mat-tab-header>
      <div
        class="mat-tab-body-wrapper"
      >
        
      </div>
    </mat-tab-group>
    
  
  </mat-card>
</pm-home>
`;
